### ISSUE TITLE
Training Classifier

### ISSUE THREAD

**Issue by akayiz** [uyuni-project/uyuni#10506]
After we obtained the preprocessed data from #10505, we need to train a classifier that, given a PR and a test it would output the probability of this test failing on the respective PR.

1. Decide how much data we will gather (i.e., how many PRs we will extract data from) (This will need experimenting with different amounts of data and monitoring the performance of the classifier, it will be a trade-off between model performance and time for data extraction and training).
2. Split the gathered data into training and testing sets. (Check work done in #10505)
3. Train different classifiers on the training data (e.g., Logistic Regression, Decision Tree, Random Forest, XGBoost) and evaluate their performance on the testing data using accuracy, precision, recall, and F1-score, and choose the best classifier for the task.
4. Experiment with feature selection and preprocessing techniques to improve model performance.
5. Train the best classifier on the whole data. (Not sure if we should do this since we won't be able to measure the performance of the resulting classifier, need to read if this is correct)
6. Export the trained classifier to be used in the test failure probability prediction program.
